Dockerfile

```diff
@@ -1,11 +1,18 @@
-FROM debian:stable
+FROM debian:stable as builder1
 RUN apt-get update && apt-get install -y maven vim build-essential gdb tree
+
 WORKDIR /build
-COPY mylibrary mylibrary
+
 COPY myapp myapp
+COPY mylibrary mylibrary
+COPY myharness myharness
+COPY pom.xml .
 
-WORKDIR /build/mylibrary
 RUN mvn install
-WORKDIR /build/myapp
-RUN mvn compile
-RUN mvn test
\ No newline at end of file
+
+FROM debian:stable
+RUN apt-get update && \
+    apt-get install -y build-essential gdb && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/*
```

Comment:
Let's not do the optimized final image.  The approach does keep image size low, but that should be the last thing for customers. It also opens up the bag of worms on `.so` not being copied in, and how to handle that, along with other issues. 

Please make this a comment with a comment stating "suggestion: reduce final image size" or remove.